src/storage-client/src/types/sources.rs

```diff
@@ -1940,6 +1892,30 @@ impl SourceConnection for PostgresSourceConnection {
     fn name(&self) -> &'static str {
         "postgres"
     }
+
+    fn upstream_name(&self) -> Option<&str> {
+        None
+    }
+
+    fn timestamp_desc(&self) -> RelationDesc {
+        RelationDesc::empty().with_column("lsn", ScalarType::UInt64.nullable(true))
+    }
+
+    fn num_outputs(&self) -> usize {
+        self.publication_details.tables.len() + 1
```

Comment:
I think the `+ 1` here and for pg is wrong? The old code doesn't have this!

Reply:
It does have it! The fact that you missed it looking at this diff tells me that splitting this trait is even more important for readability

Reply:
oh wow!